filesystem/vdi.ksy

```diff
@@ -0,0 +1,228 @@
+meta:
+  id: vdi
+  endian: le
+  encoding: utf-8
+  application:
+    - VirtualBox
+    - QEMU
+    - VMWare Workstation
+  license: GPL3.0+
+doc-ref: https://github.com/qemu/qemu/blob/master/block/vdi.c
+#  - https://github.com/qemu/qemu/blob/master/block/vdi.c
+#  - https://www.virtualbox.org/browser/vbox/trunk/src/VBox/Storage/VDI.cpp
+#  - https://forums.virtualbox.org/viewtopic.php?t=8046
```

Comment:
I'd rather put forums.virtualbox.org link here as a main one, as it is the essential source. Everything else (especially non-VirtualBox code) is reinterpretation — I'd mention them only if they contain any non-trivial clarifications / better info.

Reply:
I used QEMU source code as the main one: it is a bit simpler than vbox one. The names and docs are taken from there.  VBox source was used to look up some minor details like enum members and an additional geometry field.